src/main/java/io/vertx/core/net/impl/pool/NonBlockingSynchronization2.java

```diff
@@ -0,0 +1,43 @@
+package io.vertx.core.net.impl.pool;
+
+import io.netty.util.internal.PlatformDependent;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class NonBlockingSynchronization2<S> implements Synchronization<S> {
+
+  private final Queue<Action<S>> q = PlatformDependent.newMpscQueue(Integer.MAX_VALUE);
```

Comment:
Change this with `PlatformDependent::newMpscQueue` with no args to use an unbounded q